### ISSUE TITLE
Code checks with mypy

### ISSUE THREAD

**Issue by nickzoic** [VariantEffect/mavedb-api#126]
mypy is a very handy analysis tool which we should be using ... it'll take a bit of wrangling to get our current code to pass but it does tend to find interesting bugs (eg: #125) especially once we get typing support from our dependencies (eg: https://github.com/VariantEffect/mavehgvs/pull/36 )

**Comment by nickzoic** [comment]
now the various other branches are merged this work will continue ( https://github.com/VariantEffect/mavedb-api/tree/nick/126-code-checks-with-mypy )

**Comment by nickzoic** [comment]
OK so there's one tricky thing here, which is that there's lots of changes to models, specifically to relationship declarations, which are hard to parse out from the diffs.

So I wrote [a script](https://github.com/VariantEffect/mavedb-api/blob/nick/126-code-checks-with-mypy/tests/dump_rels.py) which dumps the details of relationships before and after a change and using that I've been able to make sure no relationships got lost and no options change.  Plus, like, the tests pass.
